aptos-move/e2e-move-tests/tests/rotate_auth_key.rs

```diff
@@ -30,6 +32,50 @@ struct RotationProofChallenge {
     new_public_key: Vec<u8>,
 }
 
+#[test]
+fn rotate_auth_key_ed25519_to_ed25519() {
+    let mut harness = MoveHarness::new();
+    let mut account1 = harness.new_account_with_key_pair();
+
+    let account2 = harness.new_account_with_key_pair();
+    // assert that the payload is successfully processed (the signatures are correct)
+    assert_successful_payload_key_rotation_from_ed25519_to_ed25519(
+        &mut harness,
+        account1.clone(),
+        *account1.address(),
+        10,
+        account2.privkey.clone(),
+        account2.pubkey.clone(),
+    );
+    // rotate account1's keypair to account2
+    account1.rotate_key(account2.privkey, account2.pubkey);
```

Comment:
@chloeqjz I thought the key rotation was called inside `assert_successful_payload_key_rotation_from_ed25519_to_ed25519` [here](https://github.com/aptos-labs/aptos-core/blob/2e9b216d0e184af6d2f8db3d78785fc9697d671f/aptos-move/e2e-move-tests/tests/rotate_auth_key.rs#L181), no?

What does this call do?

You never make this call in`rotate_auth_key_ed25519_to_multi_ed25519`...

Reply:
the key rotation function was called inside `assert_successful_payload_key_rotation_from_ed25519_to_ed25519` (this function just asserts that the key rotation on-chain is successful), but we're rotating the key in rust / off-chain (updating the keypair to the new keypair, after we successfully updated the auth key on-chain). 

updating the keypair, so that we can verify the address using `account1.auth_key()` (the new one after rotation). otherwise we'll need to pass in `account2.auth_key()` (but `account2` is only for generating a keypair in test, we don't want to create a new account just for key rotation)

Reply:
Quick summary after in-person discussion: the `account1.rotate_key(account2.privkey, account2.pubkey);` was artificial and unnecessary: we instead passed `account2.auth_key()` as an argument to `verify_originating_address`